lib/ec2imgutils/ec2setup.py

```diff
@@ -61,11 +63,21 @@ def create_security_group(self, vpc_id=None):
         group_description = 'ec2uploadimg created %s' % datetime.datetime.now()
         if not vpc_id:
             vpc_id = self.vpc_id
-        response = self._connect().create_security_group(
-            GroupName=group_name, Description=group_description,
-            VpcId=vpc_id
-        )
-
+        # Avoid name collisions
+        group_created = False
+        response = None
+        while not group_created:
+            try:
+                response = self._connect().create_security_group(
+                    GroupName=group_name, Description=group_description,
+                    VpcId=vpc_id
+                )
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'InvalidGroup.Duplicate':
+                    group_name = 'ec2uploadimg-%s' % (random.randint(1, 100))
```

Comment:
Could consolidate the two group_name generation lines and move to the beginning of for loop.

```
while not group_created:
    group_name = 'ec2uploadimg-%s' % (random.randint(1, 100))
```

Then only have:

```
if not e.response['Error']['Code'] == 'InvalidGroup.Duplicate':
    raise e
```